docs/disable-parameterized-tests.adoc

```diff
@@ -0,0 +1,250 @@
+- By an explicit `index`, starting from 0.
+- By an implicit index.
+
+Using both `name` and `index` in a single `@DisableIfParameter` annotation is not permitted.
+
+==== Targeting by `name`
+
+If naming information is included during compilation, you can target parameters by their name.
+
+[source, java]
+----
+@DisableIfParameter(name = "line2", contains = "swift")
+@ParameterizedTest
+@CsvSource({
+		"If the swift moment I entreat:;Tarry a while! You are so fair!",
+		"Then forge the shackles to my feet,;Then I will gladly perish there!"
+})
+void targetName(String line, String line2) {
+}
+----
+
+The test gets executed two times because we explicitly targeted the second parameter, which never contains the word "swift".
+
+==== Targeting by `index`
+
+You can target your parameters with their index, starting from 0.
+
+[source, java]
+----
+@DisableIfParameter(index = 1, contains = "swift")
+@ParameterizedTest
+@CsvSource({
+		"If the swift moment I entreat:;Tarry a while! You are so fair!",
+		"Then forge the shackles to my feet,;Then I will gladly perish there!"
+})
+void targetIndex(String line, String line2) {
+}
+----
+
+Again, the test gets executed two times, because we targeted the second parameter.
+
+==== Targeting by implicit index
+
+You can opt to not specify `index` or `name` and use annotation order to specify what parameter to target.
+In this case the first `@DisableIfParameter` targets the first parameter, the second annotation the second parameter, etc.
+This is mainly for convenience when you have a test method with a single parameter.
+Using this method to target parameters when your test has multiple parameters is discouraged.
+
+[source, java]
+----
+@DisableIfParameter(contains = "gibberish")
+@DisableIfParameter(contains = "gladly")
+@ParameterizedTest
+@CsvSource({
+		"If the swift moment I entreat:;Tarry a while! You are so fair!",
+		"Then forge the shackles to my feet,;Then I will gladly perish there!"
+})
+void targetByOrder(String line, String line2) {
+}
+----
+
+The test gets executed once.
+The second execution is disabled because the second argument contains "gladly".
+
+=== Using `matches`
+
+If substrings are not powerful enough, you can also use regular expressions, with the `matches` value.
+
+[source,java]
+----
+// disable invocations whose parameter ends with 'knew' or 'grew'
+@DisableIfParameter(matches = { ".*knew", ".*grew" })
+@ParameterizedTest
+@ValueSource(strings = {
+		"Lily-like, white as snow,",
+		"She hardly knew",
+		"She was a woman, so",
+		"Sweetly she grew"
+})
+void interceptMatches(String value) {
+}
+----
+
+These test invocations get disabled:
+
+* The second invocation, because it has a parameter that matches ".*knew" - ends with knew.
+* The fourth invocation, because it has a parameter that matches ".*grew" - ends with grew.
+
+Just like with `contains`, if any parameter value matches any expression from `matches`, the invocation gets disabled.
+
```

Comment:
I would add some kind of theader here, because the not is not a note for the particular example above

Reply:
I don't know what you mean. Could you clarify?

Reply:
For me I see some kind of break between line 244 (Just like..) and 246 (NOTE:) where the sentences are not connected by each other but from the formatting it seems like they belong together.